plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/TransientOAuthDataStore.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.google.cloud.tools.eclipse.appengine.login;
+
+import com.google.cloud.tools.ide.login.OAuthData;
+import com.google.cloud.tools.ide.login.OAuthDataStore;
+
+import org.eclipse.e4.core.contexts.IEclipseContext;
+
+/**
+ * Provides a transient store for saving and loading {@link OAuthData} (a user credential).
+ */
```

Comment:
Just to confirm, this is a temporary solution to storing the credentials right?

Does this support Eclipse restarting?